main/views.py

```diff
@@ -1,102 +1,118 @@
 """All views are here."""
-from django.shortcuts import render, get_object_or_404, redirect
-from .models import Post, Comment
-from .forms import PostForm, CommentForm
 from django.contrib.auth.decorators import login_required
+from django.utils.decorators import method_decorator
+from django.shortcuts import get_object_or_404, redirect
+from django.core.urlresolvers import reverse_lazy
+from django.views.generic import DetailView, ListView, CreateView, UpdateView, View, DeleteView, TemplateView
+
+from .forms import CommentForm
+from .models import Post, Comment
+
+
+class Protected(View):
+    """Protect views that need to show only for authorised users."""
 
+    @method_decorator(login_required)
+    def dispatch(self, *args, **kwargs):
+        """Rewrite standart method and it to decorator."""
+        return super().dispatch(self, *args, **kwargs)
 
-def post_list(request):
+
+class PostList(ListView):
     """Show all your Post objects."""
-    posts = Post.objects.all()
-    return render(request, 'main/index.html', {'posts': posts})
+
+    model = Post
+    context_object_name = 'posts'
+    template_name = 'main/index.html'
 
 
-def post_detail(request, pk):
+class PostDetail(DetailView):
     """Show info about one post you have chosen."""
-    post = get_object_or_404(Post, pk=pk)
-    return render(request, 'main/post_detail.html', {'post': post})
+
+    model = Post
+    template_name = 'main/post_detail.html'
 
 
-def post_new(request):
+class NewPost(CreateView):
     """Return page for adding new Post."""
-    if request.method == 'POST':
-        form = PostForm(request.POST)
-        if form.is_valid():
-            post = form.save(commit=False)
-            post.author = request.user
-            post.save()
-            return redirect('post_detail', pk=post.pk)
-    else:
-        form = PostForm()
-    return render(request, 'main/post_edit.html', {'form': form})
 
+    model = Post
+    fields = ['title', 'text']
+    success_url = reverse_lazy('post_list')
+    template_name = 'main/post_edit.html'
+
+    def form_valid(self, form):
+        """Add info to form that were not given from POST request."""
+        form.instance.author = self.request.user
```

Comment:
Try bypassing `request` object into `ModelForm`'s initializer like in https://github.com/open-craft-guild/crm-smsfly/blob/master/SMSFlyCRM/SMSApp/forms/task.py#L44-L46

You may set `form_class` instead of `model` in views :)